File: todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/ToDoApplication.java
package com.example.android.architecture.blueprints.todoapp;

import android.app.Application;
import android.os.StrictMode;

/**
 * Application class, used for setting the StrictMode.
 */
public class ToDoApplication extends Application {

    @Override
    public void onCreate() {
        if (BuildConfig.DEBUG) {
            setStrictMode();
        }
        super.onCreate();
    }

    private void setStrictMode() {
        StrictMode.setThreadPolicy(new StrictMode.ThreadPolicy.Builder()
                .detectAll()
                .penaltyLog()
                .build());

        StrictMode.setVmPolicy(new StrictMode.VmPolicy.Builder()
                .detectLeakedSqlLiteObjects()
                .penaltyLog()
                .penaltyDeath()
                .build());
    }
}

File: app/src/main/java/com/example/android/architecture/blueprints/todoapp/TodoApplication.kt
package com.example.android.architecture.blueprints.todoapp

import com.example.android.architecture.blueprints.todoapp.di.DaggerApplicationComponent
import dagger.android.AndroidInjector
import dagger.android.support.DaggerApplication
import timber.log.Timber
import timber.log.Timber.DebugTree

/**
 * An [Application] that uses Dagger for Dependency Injection.
 *
 * Also, sets up Timber in the DEBUG BuildConfig. Read Timber's documentation for production setups.
 */
open class TodoApplication : DaggerApplication() {
    override fun applicationInjector(): AndroidInjector<out DaggerApplication> {

        return DaggerApplicationComponent.factory().create(applicationContext)
    }

    override fun onCreate() {
        super.onCreate()

        if (BuildConfig.DEBUG) Timber.plant(DebugTree())
    }
}

